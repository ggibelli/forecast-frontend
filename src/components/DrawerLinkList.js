import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import ListSubheader from '@material-ui/core/ListSubheader'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemIcon from '@material-ui/core/ListItemIcon'
import ListItemText from '@material-ui/core/ListItemText'
import Collapse from '@material-ui/core/Collapse'
import InboxIcon from '@material-ui/icons/MoveToInbox'
import DraftsIcon from '@material-ui/icons/Drafts'
import SendIcon from '@material-ui/icons/Send'
import ExpandLess from '@material-ui/icons/ExpandLess'
import ExpandMore from '@material-ui/icons/ExpandMore'
import StarBorder from '@material-ui/icons/StarBorder'
import { useSelector, useDispatch } from 'react-redux'

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    maxWidth: 360,
    backgroundColor: theme.palette.background.paper,
  },
  nested: {
    paddingLeft: theme.spacing(4),
  },
}))

export default function NestedList() {
  const spotList = useSelector((state) => state.surfspots)
  const countries = spotList.map((continent) => continent.countries)
  const regions = countries.map((country) => console.log(country[0].regions))
  console.log(regions)
  // const surfSpots = regions.map(region => region.surf_spots)
  const classes = useStyles()
  const [open, setOpen] = React.useState({})

  const handleClick = (param) => {
    setOpen((prevState) => ({ ...prevState, [param]: !prevState[param] }))
  }



  return (
    <List
      component="nav"
      aria-labelledby="nested-list-subheader"
      subheader={
        <ListSubheader component="div" id="nested-list-subheader">
          World Surf Spots
        </ListSubheader>
      }
      className={classes.root}
    >
      <ListItem button onClick={() => handleClick('ciao')}>
        <ListItemIcon>
          <InboxIcon />
        </ListItemIcon>
        <ListItemText primary="Inbox" />
        {open ? <ExpandLess /> : <ExpandMore />}
      </ListItem>
      <Collapse in={open} timeout="auto" unmountOnExit>
        <List component="div" disablePadding>
          <ListItem button className={classes.nested}>
            <ListItemIcon>
              <StarBorder />
            </ListItemIcon>
            <ListItemText primary="Starred" />
          </ListItem>
        </List>
      </Collapse>

      <ListItem button onClick={handleClick2}>
        <ListItemIcon>
          <InboxIcon />
        </ListItemIcon>
        <ListItemText primary="outut" />
        {open2 ? <ExpandLess /> : <ExpandMore />}
      </ListItem>
      <Collapse in={open2} timeout="auto" unmountOnExit>
        <List component="div" disablePadding>
          <ListItem button className={classes.nested}>
            <ListItemIcon>
              <StarBorder />
            </ListItemIcon>
            <ListItemText primary="bobobo" />
          </ListItem>
        </List>
      </Collapse>
    </List>
  )
}
