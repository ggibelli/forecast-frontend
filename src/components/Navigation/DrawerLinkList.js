import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import ListSubheader from '@material-ui/core/ListSubheader'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemText from '@material-ui/core/ListItemText'
import Collapse from '@material-ui/core/Collapse'
import LinearProgress from '@material-ui/core/LinearProgress'

import ExpandLess from '@material-ui/icons/ExpandLess'
import ExpandMore from '@material-ui/icons/ExpandMore'
import { useSelector } from 'react-redux'
import { Link } from 'react-router-dom'
import updateDisplayNameToLabel from '../../utils/updateObjectsName'

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    maxWidth: 360,
    backgroundColor: theme.palette.background.paper,
  },
  nested: {
    paddingLeft: theme.spacing(4),
  },
}))

export default function NestedList() {
  const spotList = useSelector((state) => state.surfspots)
  const keysToUpdate = {
    countries: 'children',
    regions: 'children',
    surfSpots: 'children',
  }

  const nestedSpotList = updateDisplayNameToLabel(spotList, keysToUpdate)

  const classes = useStyles()
  const [open, setOpen] = React.useState({})

  const handleClick = (param) => {
    setOpen((prevState) => ({ ...prevState, [param]: !prevState[param] }))
  }
  function nestedMenu(items) {
    return items.map((nav) => {
      if (!nav.children) {
        return (
          <div key={nav.id}>
            <ListItem
              button
              className={classes.nested}
              component={Link}
              to={`/surfspots/${nav.id}`}
            >
              <ListItemText primary={nav.name} />
            </ListItem>
          </div>
        )
      }
      return (
        <div key={nav.id}>
          <ListItem button onClick={() => handleClick(nav.name)}>
            <ListItemText primary={nav.name} />
            {open[nav.name] ? <ExpandLess /> : <ExpandMore />}
          </ListItem>
          <Collapse in={open[nav.name]} timeout="auto" unmountOnExit>
            <List component="div" disablePadding>
              {nestedMenu(nav.children)}
            </List>
          </Collapse>
        </div>
      )
    })
  }

  if (nestedSpotList.length === 0) {
    return <LinearProgress />
  }

  return (
    <List
      component="nav"
      aria-labelledby="nested-list-subheader"
      subheader={
        <ListSubheader component="div" id="nested-list-subheader">
          World Surf Spots
        </ListSubheader>
      }
      className={classes.root}
    >
      {nestedMenu(nestedSpotList)}
    </List>
  )
}
