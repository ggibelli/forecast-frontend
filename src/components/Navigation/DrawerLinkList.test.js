import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import renderer from 'react-test-renderer'

import DrawerLinkList from './DrawerLinkList'

const mockStore = configureMockStore([thunk])
let store
let component

describe('DrawerLinkList', () => {
  beforeEach(() => {
    store = mockStore({
      surfspots: [
        {
          name: 'Europe',
          latitude: '12.1234',
          longitude: '34.1234',
          countries: [
            {
              name: 'cbd',
              id: '234',
              regions: [
                {
                  name: 'lol',
                  id: '090',
                  surfSpots: [
                    {
                      name: 'foz',
                      id: '456',
                    },
                  ],
                },
              ],
            },
          ],
          id: '123',
        },
      ],
    })
    component = renderer.create(
      <Provider store={store}>
        <DrawerLinkList />
      </Provider>,
    )
  })
  test('renders DrawerLinkList component', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})
