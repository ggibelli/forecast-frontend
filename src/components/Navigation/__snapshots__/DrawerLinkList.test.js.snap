// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerLinkList renders DrawerLinkList component 1`] = `
<nav
  aria-labelledby="nested-list-subheader"
  className="MuiList-root makeStyles-root-1 MuiList-padding MuiList-subheader"
>
  <div
    className="MuiListSubheader-root MuiListSubheader-sticky MuiListSubheader-gutters"
    id="nested-list-subheader"
  >
    World Surf Spots
  </div>
  <div>
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Europe
        </span>
      </div>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
    </div>
  </div>
</nav>
`;
